packages/rs-dpp/src/state_transition/state_transitions/contract/data_contract_create_transition/state_transition_estimated_fee_validation.rs

```diff
@@ -0,0 +1,53 @@
+use crate::consensus::state::identity::IdentityInsufficientBalanceError;
+use crate::fee::Credits;
+use crate::state_transition::data_contract_create_transition::accessors::DataContractCreateTransitionAccessorsV0;
+use crate::state_transition::data_contract_create_transition::DataContractCreateTransition;
+use crate::state_transition::{
+    StateTransitionEstimatedFeeValidation, StateTransitionIdentityEstimatedFeeValidation,
+    StateTransitionOwned,
+};
+use crate::validation::SimpleConsensusValidationResult;
+use crate::ProtocolError;
+use platform_version::version::PlatformVersion;
+
+impl StateTransitionEstimatedFeeValidation for DataContractCreateTransition {
+    fn calculate_min_required_fee(
+        &self,
+        platform_version: &PlatformVersion,
+    ) -> Result<Credits, ProtocolError> {
+        let base_fee = platform_version
+            .fee_version
+            .state_transition_min_fees
+            .contract_create;
+
+        let registration_cost = self
+            .data_contract()
+            .registration_cost(platform_version)
+            .unwrap_or(0);
```

Comment:
Data contract pre-checks now swallow registration_cost errors. In the old branch (feat/platformAddresses) the fee added registration_cost(platform_version)?, so any ProtocolError aborted validation (packages/rs-drive-abci/src/execution/validation/state_transition/common/validate_simple_pre_check_balance/v0/mod.rs (lines 25-104)). After the move to rs-dpp the same fee is computed with .registration_cost(...).unwrap_or(0), so unknown-version or overflow errors are converted to zero additional cost and never bubble up (packages/rs-dpp/src/state_transition/state_transitions/contract/data_contract_create_transition/state_transition_estimated_fee_validation.rs (lines 13-29) and .../data_contract_update_transition/... (lines 13-29)).

Reply:
fixed.